qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp

```diff
@@ -159,6 +159,18 @@ void ALCDataLoadingPresenter::load(const std::string &lastFile) {
   }
 
   try {
+    if (lastRunPath.toString() == "") {
+      throw std::invalid_argument(
```

Comment:
Instead of trying and catching in the same method (which means we pay several costs in readability / runtime for little benefits)

Could you do

```
if (lastRunPath.toString() == ""){
    m_view->displayError("The last run is now a ...");
    m_view->enableAll();
    m_loadingData = false;
    return;
}

if (firstRun.toString() == ""){
    m_view->displayError("The last run is now a ...");
    m_view->enableAll();
    m_loadingData = false;
    return;
}
```

As it highlights in the future we should refactor this method to do something like

```
void ALCDataLoadingPresenter::loadWrapper(...){
     m_loadingData = true;
     m_view->disableAll();

     load();  // Existing impl

     m_loadingData = false;
     m_view->enableAll();
```

Rather than (ab)using exceptions for the same effect